extensions/binding/org.eclipse.smarthome.binding.ble/src/main/java/org/eclipse/smarthome/binding/ble/BleBridgeApi.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.ble;
+
+/**
+ * The {@link BleBridgeApi} class defines the standard bridge API that must be implemented by Thing implemented in the
+ * BLE host.
+ * <p>
+ * <b>Scanning</b>
+ * The API assumes that the bridge is "always" scanning to enable beacons to be received.
+ * The bridge must decide to enable and disable scanning as it needs. This design choice avoids interaction between
+ * higher layers where a binding may want to enable scanning while another needs to disable scanning for a specific
+ * function (e.g. to connect to a device). The bridge should disable scanning only for the period that is needed.
+ *
+ * @author Chris Jackson - Initial contribution
+ */
+public interface BleBridgeApi {
+
+    /**
+     * Starts an active scan on the Bluetooth interface.
+     *
+     * @return true if the scan was started
+     */
+    boolean scanStart();
```

Comment:
> I expected that I could register some callback(or listener) that will be called(notified) when a scan result is found.

Why do you want this callback? This is really only for discovery. Scanning should be enabled at all times - I'm not sure if I say this above or in the code of the bridge, but this is generally needed (at least in BlueGiga) in order to get the beacons at all. The startScan method is used for discovery only, and enables active scanning.

Are you looking for the beacons? If so, then all you need to do is implement BleDeviceListener.onScanRecordReceived() callback to receive the results of any records received.

If not, can you describe what you want to do and I'll try and answer...